src/economy/expedition_bootstrap.cc

```diff
@@ -57,9 +57,20 @@ void ExpeditionBootstrap::is_ready(Game& game) {
 
 // static
 void ExpeditionBootstrap::input_callback(
-   Game& game, InputQueue*, DescriptionIndex, Worker*, void* data) {
+   Game& game, InputQueue* queue, DescriptionIndex, Worker*, void* data) {
 	ExpeditionBootstrap* eb = static_cast<ExpeditionBootstrap*>(data);
 	eb->is_ready(game);
+	// If we ask for several additional items of the same type, it may happen that a
+	// specific item was originally requested by queue B but is put into queue A. This
+	// causes both queues to cancel their requests so that some transfers are
+	// accidentally cancelled. The solution is to iterate ALL queues of this type and
+	// check whether their count and transfers still match up.
+	for (auto& pair : eb->queues_) {
```

Comment:
I'm not a fan of `auto`. My editor is too dumb to figure out the type so I have to do it myself.